Add pre and post migration scripts for mail

bwrsandman opened the pull request:
~~Depends on #13~~

Commit:
Init of analysis work file for mail

Signed-off-by: Sandy Carter <sandy.carter@savoirfairelinux.com>

gurneyalex committed:
First part of product migration analysis.
XML ID migration complete

gurneyalex committed:
fix missing dependency in product in v8

gurneyalex committed:
updated product migration analysis

gurneyalex committed:
first pass at product migration script

gurneyalex committed:
fix bad refactoring

gurneyalex committed:
compute active attribute of template from attribute of variants

gurneyalex committed:
copy product_product.image to product_product.image_variant

gurneyalex committed:
updated work analysis

hbrunn commented on addons/mail/migrations/8.0.1.0/post-migration.py:
```diff
@@ -0,0 +1,68 @@
+    legacy_reply_to = openupgrade.get_legacy_name('reply_to')
+    mail_ids = mail_mail_pool.search(
+        cr, uid, [])
+    for mail_obj in mail_mail_pool.browse(cr, uid, mail_ids):
```
why don't you simply update from a join over mail_message and mail_mail? The way you do it currently is very expensive I think and can really cause trouble for a lot of mails/messages

bwrsandman replied:
Will do that.
Thanks

gurneyalex committed:
migration of product.packaging and product.ul

guewen committed:
migration of product_template.cost_method, moved in stock_account

bwrsandman commented:
@hbrunn Replaced loop with SQL query for greater efficiency.

gurneyalex committed:
reinstall data from product moved to mrp

gurneyalex committed:
add migration for product.packaging to template

gurneyalex committed:
fix error in previous refactoring

gurneyalex committed:
Merge branch 'product-v8-migration' of https://github.com/guewen/OpenUpgrade into product-v8-migration

guewen committed:
Migrate product_template.standard_price.

Now a fields.property so the prices are moved to properties.
Also, entries for the prices will be created in product_price_history.

legalsylvain committed:
[ADD] check value for ir_ui_view.type field.

legalsylvain committed:
[REF] Remove useless field migration because 'oldname' is defined;

guewen committed:
Updated openupgrade_analysis_WORK.txt with comments

gurneyalex committed:
packaging field is o2m -> no colum renaming required

gurneyalex committed:
fix packaging/ul migration

gurneyalex committed:
re-add the change on product consultant

gurneyalex committed:
added AGPL header

gurneyalex committed:
fix typo

guewen committed:
Merge remote-tracking branch 'openupgrade/master' into product-v8-migration

guewen committed:
openerp namespace is missing in the import

guewen committed:
Wrong API call (PR submitted upstream)

guewen committed:
Migrate 'variants' and 'price_extra' using the new attribute models

legalsylvain committed:
[IMP] 'base' analysis. now 'ir.actions.server' changes only are missing

guewen committed:
Merge remote-tracking branch 'gurneyalex/product-v8-migration' into product-v8-migration

Conflicts:
	addons/product/migrations/8.0.1.1/post-migration.py

guewen committed:
pep8, unused models

hbrunn commented:
:+1:

guewen committed:
Write a 'Unix time zero' on the product price history entries so product have a standard price for the past, not only from now

florentx committed:
[REF] openupgrade_records: sorting key can be a tuple

legalsylvain committed:
[ADD] finish analyses of 'base';
[ADD] post-migration files to manage invalid state value for ir_actions_server

arthru commented on addons/mail/migrations/8.0.1.0/openupgrade_analysis_work.txt:
```diff
@@ -0,0 +1,41 @@
+
+mail         / mail.group               / message_last_post (datetime)  : NEW
+mail         / res.partner              / message_last_post (datetime)  : NEW
+## TODO something with last_post in openupgrade80
```
This TODO looks like there is work needed, but you treated it in post_migration script. Could you change that message ?

guewen committed:
pep8: missing whitespace around operator

legalsylvain committed:
[REF] Pep8

guewen committed:
Migrate product_template.cost_method to property (in stock_account)

bwrsandman commented:
@arthru Good catch!
That's fixed now.

paulius-sladkevicius commented:
:+1:

legalsylvain committed:
Remove a line in order to be pep8 compliant after merge

arthru committed:
Merge branch 'base' of https://github.com/legalsylvain/OpenUpgrade into legalsylvain-base

Conflicts:
	openerp/addons/base/migrations/8.0.1.3/pre-migration.py

legalsylvain commented on addons/mail/migrations/8.0.1.0/openupgrade_analysis_work.txt:
```diff
@@ -0,0 +1,41 @@
+mail         / mail.alias               / alias_contact (selection)     : NEW required: required, selection_keys: ['everyone', 'followers', 'partners'], req_default: everyone
+mail         / mail.alias               / alias_parent_model_id (many2one): NEW relation: ir.model
+mail         / mail.alias               / alias_parent_thread_id (integer): NEW 
+mail         / mail.mail                / recipient_ids (many2many)     : NEW relation: res.partner
```
[need information, code review : test and review]
Is it normal if alias_parent_model_id (and alias_parent_thread_id) are not initialized ?
I tested realizing diff between 7.0 database migrated to 8.0 and 8.0 just-installed and the data of those 2 columns are not the same.
(in migrated database the two fields are empty)

hbrunn replied:
I wouldn't know how to fill this columns during migration. I see that only used in hr_reruitment and crm, this modules will have to update their data accordingly.

arthru committed:
Merge pull request #47 from florentx/sort_key

[REF] openupgrade_records: sorting key can be a tuple

legalsylvain committed:
Merge pull request #60 from arthru/legalsylvain-base

Base migration finished, based on work from legalsylvain

Commit:
Merge pull request #37 from guewen/product-v8-migration

[ADD] Product v8 migration

bwrsandman commented:
Moving target branch

Commit:
Add pre and post migration scripts for mail

Migrate selections for notifications to new selection options
Move 'server_id' and 'reply_to' from mail.mail to mail.message

Signed-off-by: Sandy Carter <sandy.carter@savoirfairelinux.com>
